src/main/java/rs/teslaris/core/service/impl/document/strategydecorator/ConferenceJPAServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package rs.teslaris.core.service.impl.document.strategydecorator;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Component;
+import rs.teslaris.core.model.document.Conference;
+import rs.teslaris.core.repository.document.ConferenceRepository;
+import rs.teslaris.core.service.impl.JPAServiceImpl;
+
+@Component
+public class ConferenceJPAServiceImpl extends JPAServiceImpl<Conference> {
```

Comment:
In which cases decorator pattern is applied? For example, why not for JournalServiceImpl?

Reply:
It is used only for hierachically decoupled services. Because of the fact that, in inheroitors, we must inherit the base service which leaves no more room to inherit JpaServiceImpl<>.

Reply:
There are other possible workarounds that @milospp presented, but i believe this is the best one that we have come up right now.